data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -746,4 +748,48 @@ public SafeFuture<Optional<UInt64>> getFinalizedStateSlot(final UInt64 beforeSlo
         .getLatestAvailableFinalizedState(beforeSlot)
         .thenApply(maybeState -> maybeState.map(BeaconState::getSlot));
   }
+
+  public SafeFuture<Optional<ObjectAndMetaData<SszList<PendingPartialWithdrawal>>>>
+      getPendingPartialWithdrawals(final String stateIdParam) {
+    return stateSelectorFactory
+        .createSelectorForStateId(stateIdParam)
+        .getState()
+        .thenApply(this::getPendingPartialWithdrawals);
+  }
+
+  private Optional<ObjectAndMetaData<SszList<PendingPartialWithdrawal>>>
+      getPendingPartialWithdrawals(final Optional<StateAndMetaData> maybeStateAndMetadata) {
+    checkMinimumMilestone(maybeStateAndMetadata, SpecMilestone.ELECTRA);
+
+    return maybeStateAndMetadata.map(
+        stateAndMetaData -> {
+          final SszList<PendingPartialWithdrawal> withdrawals =
+              stateAndMetaData
+                  .getData()
+                  .toVersionElectra()
+                  .orElseThrow()
+                  .getPendingPartialWithdrawals();
+          return new ObjectAndMetaData<>(
+              withdrawals,
+              stateAndMetaData.getMilestone(),
+              stateAndMetaData.isExecutionOptimistic(),
+              stateAndMetaData.isCanonical(),
+              stateAndMetaData.isFinalized());
+        });
+  }
+
+  private void checkMinimumMilestone(
+      final Optional<StateAndMetaData> maybeStateAndMetadata,
+      final SpecMilestone minimumMilestone) {
+    if (maybeStateAndMetadata.isPresent()
+        && maybeStateAndMetadata
+            .get()
+            .getMilestone()
+            .isLessThanOrEqualTo(minimumMilestone.getPreviousMilestone())) {
```

Comment:
NIT: it would be great to add isLessThan() to Milestone... :)

`.isLessThan(minimumMilestone)` is easier to read